src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -596,8 +596,13 @@ private static List<Content> appendManifestContents(
 
         manifest.append("Requested components:\n\n");
         ContentContainer contentsContainer = new ContentContainer(contentFilter, components);
+        Set<Class<? extends Component>> supersededComponents = new HashSet<>();
+        components.forEach(c -> supersededComponents.addAll(c.getSupersededComponents()));
         for (Component component : components) {
             try {
+                if (supersededComponents.contains(component.getClass())) {
```

Comment:
The use of `Class` in APIs is usually a bit of an antipattern since it makes assumptions about implementation hierarchies; and actually `@Extension` is not strictly limited to singleton classes. Maybe a `boolean supersedes(Component)`?

Reply:
Right. Some components have class Hierarchy, for example [ProcFilesRetriever](https://github.com/jenkinsci/support-core-plugin/blob/1538.v0a_a_a_e0f2e1c0/src/main/java/com/cloudbees/jenkins/support/impl/ProcFilesRetriever.java).

You mean a `Component#supersedes(component)` ? That would require a O(n^2) loop to populated the list of superseded components ? But I guess should be fine, we are not talking about 100s of components.